packages/middleware-sdk-transcribe-streaming/src/websocket-handler.ts

```diff
@@ -0,0 +1,188 @@
+import { HttpHandlerOptions, RequestHandlerMetadata } from "@aws-sdk/types";
+import { HttpHandler, HttpRequest, HttpResponse } from "@aws-sdk/protocol-http";
+import { formatUrl } from "@aws-sdk/util-format-url";
+
+export interface WebSocketHandlerOptions {
+  /**
+   * The maximum time in milliseconds that the connection phase of a request
+   * may take before the connection attempt is abandoned.
+   */
+  connectionTimeout?: number;
+}
+
+/**
+ * Base handler for websocket requests. By default, the request input and output
+ * body will be in a ReadableStream, because of interface consistency among middleware.
+ * If ReadableStream is not available, like in React-Native, the response body
+ * will be an async iterable.
+ */
+export class WebSocketHandler implements HttpHandler {
+  public readonly metadata: RequestHandlerMetadata = {
+    handlerProtocol: "websocket"
+  };
+  private readonly connectionTimeout: number;
+  constructor({ connectionTimeout }: WebSocketHandlerOptions = {}) {
+    this.connectionTimeout = connectionTimeout || 2000;
+  }
+
+  destroy(): void {}
+
+  handle(
+    request: HttpRequest,
+    options: HttpHandlerOptions = {}
+  ): Promise<{ response: HttpResponse }> {
+    const url = formatUrl(request);
+    const socket: WebSocket = new WebSocket(url);
+    socket.binaryType = "arraybuffer";
+    return waitForReady(socket, this.connectionTimeout)
+      .then(() => {
+        const { body } = request;
+        const bodyStream = getIterator(body);
+        return connect(socket, bodyStream);
+      })
+      .then(iterableToReadableStream)
+      .then(bodyStream => ({
+        response: new HttpResponse({
+          statusCode: 200, // indicates connection success
+          body: bodyStream
+        })
+      }));
+  }
```

Comment:
Function can be simplified using async-await
```suggestion
  async handle(
    request: HttpRequest,
    options: HttpHandlerOptions = {}
  ): Promise<{ response: HttpResponse }> {
    const url = formatUrl(request);
    const socket: WebSocket = new WebSocket(url);
    socket.binaryType = "arraybuffer";
    await waitForReady(socket, this.connectionTimeout);
    const { body } = request;
    const bodyStream = getIterator(body);
    const asyncIterable = connect(socket, bodyStream);
    const bodyStream_1 = iterableToReadableStream(asyncIterable);
    return ({
      response: new HttpResponse({
        statusCode: 200,
        body: bodyStream_1
      })
    });
  }
```

I also removed unnecessary await for `iterableToReadableStream` in above suggestion